code/controllers/subsystem/overlays.dm

```diff
@@ -155,37 +150,30 @@ SUBSYSTEM_DEF(overlays)
 
 	overlays = build_appearance_list(overlays)
 
-	LAZYINITLIST(our_overlays)	//always initialized after this point
+	LAZYINITLIST(add_overlays) //always initialized after this point
 	LAZYINITLIST(priority_overlays)
-
-	var/list/cached_overlays = our_overlays	//sanic
-	var/list/cached_priority = priority_overlays
-	var/init_o_len = cached_overlays.len
-	var/init_p_len = cached_priority.len  //starter pokemon
-	var/need_compile
+	LAZYINITLIST(remove_overlays)
 
 	if(priority)
-		cached_priority += overlays  //or in the image. Can we use [image] = image?
-		need_compile = init_p_len != cached_priority.len
+		priority_overlays += overlays  //or in the image. Can we use [image] = image?
```

Comment:
this doesn't make any sense, why would you add all overlays to the priority list

Reply:
It already did that

Reply:
overlays in this context is a proc arg, not a atom var

Reply:
oI see, that is confusing